### PR TITLE
[Qt6] Fix crash with Python on QgsLocatorResult::userData (fixes #56456)

### LINKED ISSUE #56456: LTR - posgresql provider is unable to get featureCount() until certain conditions - pg_class.reltuples isn't populated yet
### What is the bug or the crash?

After digging hours within an strange problem at qgis-server, I finally found where/why it happens.

While I'm opening this issue to inform, I would need some guidance on how to provide you the concrete steps to reproduce this in order to be fixed.

The way I discovered this was through an unexpected response from qgis-server's `OGC WFS3 API` handler `getFeatures`, but It seems related to the Postgres provider.

The exact place is while querying the feature count at <https://github.com/qgis/QGIS/blob/252ad49ddcbc4a0dcfe9eb9381503de0fde9e0ed/src/providers/postgres/qgspostgresprovider.cpp#L3794>, using a [well-known method](https://stackoverflow.com/a/7945274/1894803) as is:

```sql
SELECT reltuples::bigint FROM pg_catalog.pg_class WHERE oid=regclass('"test_schema"."test_table"')::oid;
```

**The problem with this query is it may return a count of `-1` until the pg_class stats are populated**

qgis-server uses this feature count to generate a range to validate the `offset` parameter, which can't be in the `0..-1` range, such failing to answer with:

```plain
12:04:16 WARNING Server[20]: [{"code":"Bad request error","description":"Argument 'offset' is not valid. Offset for features to retrieve [0--1]"}]
```

A workaround may be manually triggering an [`ANALYZE`](https://www.postgresql.org/docs/current/sql-analyze.html) within affected table, i.e.

```sql
ANALYZE VERBOSE test_schema.test_table;
```

### Steps to reproduce the issue

Please read above

### Versions

```bash
/usr/lib/cgi-bin/qgis_mapserv.fcgi --version
```

```plain
QGIS 3.28.15-Firenze 'Firenze' (exported)
QGIS code branchRelease 3.28
Qt version 5.15.10
Python version 3.11.7
Compiled against GDAL/OGR 3.8.3
Running against GDAL/OGR 3.8.4
PROJ version 9.3.1
EPSG Registry database version v10.098 (2023-11-24)
GEOS version 3.12.1-CAPI-1.18.1
Compiled against SQLite 3.45.0
Running against SQLite 3.45.1
OS Debian GNU/Linux trixie/sid
```

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by rouault**
Credits to @nyalldawson for the hints how to fix

### CODE REVIEW

**Comment by rouault** [comment]
should we backport qt6 related fixes to 3.36 ?

**Comment by nyalldawson** [comment]
@rouault 

> should we backport qt6 related fixes to 3.36 ?

I think so, if they aren't too intrusive. I'm still hoping we get unofficial/preview Qt6 builds of QGIS 3.36 for wider testing.

File: src/core/locator/qgslocatorfilter.h [hunk 1]
```diff
@@ -57,14 +57,14 @@ class CORE_EXPORT QgsLocatorResult
      *
      * \since QGIS 3.18
      */
-    QVariant getUserData() const;
+    QVariant getUserData() const SIP_PYNAME( userData );
```
Comment (nyalldawson) [inline src/core/locator/qgslocatorfilter.h:60]:
```suggestion
    QVariant userData() const SIP_PYNAME( _userData );
```

Comment (rouault) [reply src/core/locator/qgslocatorfilter.h]: done, and related .cpp changes to take into account the method renaming

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**